### ISSUE TITLE
On the join block modal review accessibility

### ISSUE THREAD

**Issue by jaymcp** [amnestywebsite/humanity-theme#30]
**Describe the bug**  
On the join block modal review accessibility

**To Reproduce**  
1. Go to https://www.amnesty.org/en/get-involved/join/
2. Tab to Join Now
3. See focus

**Expected behaviour**  
We should trap the focus within the modal when activated via keyboard or screen reader

**Actual behaviour**  
The join modal is lacking in accessibility support. Specifically, it doesn't trap focus within the modal when activated via keyboard or screen reader.

**Link to a page where the bug is reproduced in isolation (optional)**  
https://www.amnesty.org/en/get-involved/join/

**Screenshots**  
![image](https://github.com/amnestywebsite/amnesty-wp-theme/assets/37809410/1d2422b9-66de-4639-bbd9-ce75fd43c0ab)

**Comment by Gentyspun** [comment]
PR Created: https://github.com/amnestywebsite/wp-plugin-theme-companion/pull/61

**Comment by jaymcp** [comment]
This is currently blocked by https://github.com/amnestywebsite/wp-plugin-theme-companion/issues/67

**Comment by jaymcp** [comment]
closing as complete via https://github.com/amnestywebsite/amnesty-wp-theme/issues/2904
